Mathlib/NumberTheory/Modular.lean

```diff
@@ -231,14 +231,13 @@ theorem smul_eq_lcRow0_add {p : Fin 2 → ℤ} (hp : IsCoprime (p 0) (p 1)) (hg
   have nonZ1 : (p 0 : ℂ) ^ 2 + (p 1 : ℂ) ^ 2 ≠ 0 := mod_cast hp.sq_add_sq_ne_zero
   have : ((↑) : ℤ → ℝ) ∘ p ≠ 0 := fun h => hp.ne_zero (by ext i; simpa using congr_fun h i)
   have nonZ2 : (p 0 : ℂ) * z + p 1 ≠ 0 := by simpa using linear_ne_zero z this
-  simp only [coe_specialLinearGroup_apply, hg, algebraMap_int_eq, Int.coe_castRingHom,
-    Complex.ofReal_intCast]
-  field_simp [nonZ1, nonZ2, denom_ne_zero]
-  rw [(by simp :
-    (p 1 : ℂ) * z - p 0 = (p 1 * z - p 0) * ↑(Matrix.det (↑g : Matrix (Fin 2) (Fin 2) ℤ)))]
-  rw [← hg, det_fin_two]
-  simp only [Int.cast_sub, Int.cast_mul]
-  ring
+  subst hg
+  rw [coe_specialLinearGroup_apply]
+  replace nonZ2 : z * (g 1 0 : ℂ) + g 1 1 ≠ 0 := by convert nonZ2 using 1; ring
+  simp [field]
+  have H := congr(Int.cast (R := ℂ) $(det_fin_two g))
+  simp at H
+  linear_combination -((z : ℂ) * (g 1 1 : ℂ) - g 1 0) * H
```

Comment:
```suggestion
  have H := congr(Int.cast (R := ℂ) $(det_fin_two g))
  simp at H
  simp [field]
  linear_combination -((z : ℂ) * (g 1 1 : ℂ) - g 1 0) * H
```

Reply:
`grind` also works here in place of `linear_combination`, but if there's information in the linear_combination call that's fine.

Reply:
I accepted the first suggestion; I don't have an opinion on preferring linear_combination or grind.